hb-coq-vernac.elpi

```diff
@@ -0,0 +1,95 @@
+/*        Hierarchy Builder: algebraic hierarchies made easy
+    This software is released under the terms of the MIT license              */
+
+% This file contains proxies for the API to write into the Coq state (eg add
+% an inductive, set implicit arguments...). This has two purposes:
+% - give nicer error messages
+% - we want to be able to log to a file all these actions (WIP), this file is
+%   were logging takes place. Hence we never call, say, coq.evn.add-* directly,
+%   but always via this proxy
+
+namespace coq.vernac {
+
+pred set-implicit i:bool, i:gref, i:list implicit_kind.
+set-implicit Global? GR I :-  std.do! [
+  if (Global? = tt)
+    (@global! => coq.arguments.set-implicit GR [I])
+    (            coq.arguments.set-implicit GR [I]),
+].
+
+pred add-const i:id, i:term, i:term, i:opaque?, o:constant.
+add-const Name Bo Ty Opaque C :- std.do! [
+  % TODO: refine when we switch to add-section-variable/add-const
+  if (not(ground_term Ty ; ground_term Bo))
+    (coq.error "HB: cannot infer some information in" Name
+               ":" {coq.term->string Ty} ":=" {coq.term->string Bo})
+    true,
+  coq.env.add-const Name Bo Ty Opaque C,
+  % if (var Ty) (Ty? = none) (Ty? = some Ty),
+  set-implicit ff (const C) [],
+].
+
+pred add-variable i:id, i:term, o:constant.
```

Comment:
Isn't it supposed to be `add-section-variable`?

Reply:
ok

Reply:
I'm doing this atop of the other PR